internal/elasticsearch/client.go

```diff
@@ -182,6 +181,13 @@ func sendEsRequest(cluster, namespace string, payload *EsRequest, client k8sclie
 	// we use the insecure TLS client here because we are providing the SA token.
 	httpClient := getTLSClient(cluster, namespace, client)
 	resp, err := httpClient.Do(request)
+	if err != nil {
+		if resp == nil {
+			payload.Error = err
+			return
+		}
+	}
```

Comment:
You’re not handling the error unless resp is nil

Reply:
right, because if it isn't nil we may be retrying our request with a different type of client